pyuvdata/uvdata/miriad.py

```diff
@@ -41,1486 +41,1529 @@ def _pol_to_ind(self, pol):
-                    self.telescope_location_lat_lon_alt = (
-                        latitude,
-                        longitude,
-                        telescope_obj.telescope_location_lat_lon_alt[2],
-                    )
-                if lat_close and lon_close:
-                    if correct_lat_lon:
-                        warnings.warn(
-                            "Altitude is not present in Miriad file, "
-                            "using known location values for "
-                            "{telescope_name}.".format(
-                                telescope_name=telescope_obj.telescope_name
-                            )
-                        )
-                    else:
-                        warnings.warn(
-                            "Altitude is not present in Miriad file, "
-                            "using known location altitude value "
-                            "for {telescope_name} and lat/lon from "
-                            "file.".format(telescope_name=telescope_obj.telescope_name)
-                        )
-                else:
-                    warn_string = "Altitude is not present in file "
-                    if not lat_close and not lon_close:
-                        warn_string = (
-                            warn_string
-                            + "and latitude and longitude values do not match values "
-                        )
-                    else:
-                        if not lat_close:
-                            warn_string = (
-                                warn_string + "and latitude value does not match value "
-                            )
-                        else:
-                            warn_string = (
-                                warn_string
-                                + "and longitude value does not match value "
-                            )
-                    if correct_lat_lon:
-                        warn_string = (
-                            warn_string + "for {telescope_name} in known "
-                            "telescopes. Using values from known "
-                            "telescopes.".format(
-                                telescope_name=telescope_obj.telescope_name
-                            )
-                        )
-                        warnings.warn(warn_string)
-                    else:
-                        warn_string = (
-                            warn_string + "for {telescope_name} in known "
-                            "telescopes. Using altitude value from known "
-                            "telescopes and lat/lon from "
-                            "file.".format(telescope_name=telescope_obj.telescope_name)
-                        )
-                        warnings.warn(warn_string)
+            # Need the array direction here since channel_width is always supposed
+            # to be > 0, but channels can be in decending freq order
+            if self.future_array_shapes:
```

Comment:
The if/else logic seems to lead to the same outcome -- the math operation appears to be the same. Is something missing here?

Reply:
Oops, yes. The second one should have be `freq_dir = np.sign(np.diff(freq_array_use[0, ([0, -1])]))` since the freq_array has 2 dimensions. Not sure why that didn't cause a problem in tests.

Reply:
oh, I see. This isn't needed because I make it 1D in either case earlier in the code.